module-2/backend/week_four.md

```diff
@@ -10,14 +10,26 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 * What is a cookie?
+  - A cookie is a piece of information that is stored on a client's (someone who visits a website) computer that persists between sessions.
 * What’s the difference between a session and a cookie?
+  - A session can be deleted by a web application, while cookies need to be deleted by the client.
 * What’s a flash and when do you want to use flashes?
+  - A flash is a rails hash construct that allows you to temporarily display a message in response to a condition.
 * Why do people say “HTTP is stateless”?
+  - HTTP is a protcol and does not hold a state.  This is similar to the idea of a ruby module being stateless in the sense that there is no variable or other container that holds a state.
 * What’s authentication? Explain.
+  - Authentication is the mechanism by which a website can identify a particular user and verify that they are who they say they are.
 * What’s the difference between authentication and authorization?
+  - Authorization is the term for restricting access to information or web pages based upon a user's identity.  Based on a user's identity they may be allowed or disallowed to visit particular places in a website.
 * What’s a before filter?
+  - A before filter allows you to set helper methods that occur before a controller action is initiated (such as "set_backpack" to set up a new backpack before each controller action)
 * How do we keep track of a user once they’ve logged in?
+  - Through the use of the session hash.  This can store a user ID, a user name and other information to keep track of which user is logged in from a particular computer.
 * When do you want to namespace a resource? When do you want to nest a resource? What's the differences between those two approaches?
+  - A namespaced resource allows you to set up a particular area of a website that allows an admin or other type of superclass of user to access a particular section of a website.  This allows you to control the authorization for a large number sub-pages from a single place.  A nested resource is for sub-resources that allow you to generate nested pages, such as categories/food.  This would let you see particular categories of food. 
 * At a high level, what tools can you use to implement authorization? How would you use them?
+  - An encryption / hashing algorithm of some kind such as bcrypt can allow you to securely store password.  The other thing that can be used is a gem such as devise or clearance to generate the authentication mechanisms.  I appreciate that we are learning how to do it manually, but feel like with security-related things it is generally better to use a tested solution.
```

Comment:
We're using BCrypt to store our hashed passwords, but we're using sessions/before filters/inheritance to do our authorization. I tend to agree when it comes to tools we have access to for auth, but these are still tools that will be good to have in your toolbox. Not necessarily limited to auth, even though that's what we're using them to accomplish now.